File: PaED62__cpp/PaED62.cpp
#include <iostream>

std::string hash(std::string sInput)
{
	std::string sOutput = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";

	for (int j = 0; j < sInput.size(); ++j)
		for (int i = 0; i < sOutput.size(); ++i)
			sOutput[i] = (char)((((int)sOutput[i] * (i + 1)) + ((int)sInput[j] * (j + 1))) % (126 - 33) + 33);

	return sOutput;
}

int main(int argc, char const *argv[])
{
	std::cout << hash("Hello, world!") << std::endl;
	std::cout << hash("Hello, world! ") << std::endl;

	system("pause > nul");
	return 0;
}

File: PaED62__js/PaED62.js
const hash = (sInput) =>
{
	sOutput = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz".split("");

	for (let j = 0; j < sInput.length; ++j)
		for (let i = 0; i < sOutput.length; ++i)
			sOutput[i] = String.fromCharCode(((sOutput[i].charCodeAt(0) * (i + 1)) + (sInput.charCodeAt(j) * (j + 1))) % (126 - 33) + 33);

	return sOutput.join("");
};

console.log(hash("Hello, world!"));
console.log(hash("Hello, world! "));